mono/metadata/object.c

```diff
@@ -725,8 +725,8 @@ compute_class_bitmap (MonoClass *class, gsize *bitmap, int size, int offset, int
 			if (field->type->byref)
 				break;
 
-			if (static_fields && field->offset == -1)
-				/* special static */
+			// Special static fields do not have a domain-level static slot
```

Comment:
No, the bug here was that we expected the -1 optimization to have kicked in this early, but it actually hadn't, and probably never would. We compute regular static field bitmaps before dealing with special static fields, and the latter part is where we apply the optimization. (See `mono_class_create_runtime_vtable ()`.)

In any case, with 0686c213583a70bde97c8d04e0748e47416bfaf9, `mono_class_field_is_special_static ()` will take advantage of the optimization when it's possible, so now we may as well use that instead of the somewhat cryptic -1 check.